src/components/Input/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import { InputProps } from './types';
+
+import OptimizedImage from 'components/OptimizedImage';
+
+import EyeIcon from '../../../public/images/icons/eye.svg';
+import EyeSlashIcon from '../../../public/images/icons/eye-slash.svg';
+
+const Input: Component<InputProps> = ({
```

Comment:
Usando composição, cria um componente `InputBase` onde tenha a base do estilo do input e algumas regras importantes, depois cria o `PasswordInput` e lá você adiciona as lógicas do input de senha.

Depois exporta eles individualmente, além de deixar o export default sendo um outro componente `Input`, onde recebe um type e nele tu bota condicional para exibir ou o input password se for type password, ou o input base se for outro tipo